tokio-util/src/io/sync_bridge.rs

```diff
@@ -0,0 +1,46 @@
+use pin_project_lite::pin_project;
+use std::io::{Read, Write};
+use std::pin::Pin;
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
+
+pin_project! {
+/// Use a [`tokio::io::AsyncRead`] synchronously as a [`std::io::Read`] or
+/// a [`tokio::io::AsyncWrite`] as a [`std::io::Write`].
+#[derive(Debug)]
+pub struct SyncIOBridge<T> {
+    src: Pin<Box<T>>,
+    #[pin]
+    rt: tokio::runtime::Handle,
+}
+}
+
+impl<T: AsyncRead> Read for SyncIOBridge<T> {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        let src = &mut self.src;
+        self.rt.block_on(src.read(buf))
+    }
+}
+
+impl<T: AsyncWrite> Write for SyncIOBridge<T> {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let src = &mut self.src;
+        self.rt.block_on(src.write(buf))
+    }
+
```

Comment:
we may want to forward other methods, like `write_all`, so that we can avoid the overhead of multiple `block_on` calls if we have to make multiple writes?

Reply:
similarly, we probably want to have some way of forwarding `write_vectored` calls; otherwise, we'll lose vectored writes even if the underlying `AsyncWrite` resource supports vectored writes. it's a shame that the `std::io::Write` version of `is_write_vectored` is unstable, but we could also add a

```rust
impl<T: AsyncWrite> SyncIoBridge<T> {
    pub fn is_write_vectored(&self) -> bool {
         self.src.is_write_vectored()
    }
}
```
or something?

Reply:
Ah weird, and it's unstable just because of a need to potentially bikeshed the name?   Seems...unnecessary.

Anyways, applied that suggestion, thanks!